libsolidity/analysis/ContractLevelChecker.cpp

```diff
@@ -621,58 +621,121 @@ void ContractLevelChecker::checkBaseABICompatibility(ContractDefinition const& _
 
 void ContractLevelChecker::checkAmbiguousOverrides(ContractDefinition const& _contract) const
 {
-	vector<FunctionDefinition const*> contractFuncs = _contract.definedFunctions();
-
-	auto const resolvedBases = resolveDirectBaseContracts(_contract);
-
-	FunctionMultiSet inheritedFuncs = inheritedFunctions(&_contract);;
+	// Fetch non overridden base functions and sort them by signature.
+	FunctionMultiSet nonOverriddenBaseFunctionSet = convertContainer<FunctionMultiSet>(
```

Comment:
Why do you stil need this?

Reply:
I mean shouldn't it rather return the graph instead of removing overridden base functions?

Reply:
I need functions to connect the current contract to initially to construct the graph - that's exactly what this function returns. Note that it no longer removes overridden base functions, but only doesn't add base functions of base functions, if they were already overridden.

Reply:
At least that's the idea - I may need to adjust slightly...

Reply:
It was in fact still slightly wrong - I just don't want to propagate the non-overridden base functions of a base, if they are overridden in that same base - I changed it now. As a matter of fact I even had a test for that case, but didn't notice, because it had wrong expectations (also adjusted in last commit).